### PR TITLE
Quick Action Menus: Add Karma Check for Accessibility Violations 

### LINKED ISSUE #7751: Quick Action Menus: Add aXe karma tests
## Feature Description

Per discussion here: https://github.com/google/web-stories-wp/issues/7530#issuecomment-842498164 we're going to use this new feature to try out adding more aXe coverage.

## Acceptance Criteria

- Update quick action menu karma tests to check for accessibility using aXe (fix anything that's found!)

### PR DESCRIPTION
**Pull request by BrittanyIRL**
## Context

Per discussion here: https://github.com/google/web-stories-wp/issues/7530#issuecomment-842498164 starting to add in aXe checks for accessibility violations.

## Summary

Tweak the context menu for accessibility testing. 

## Relevant Technical Choices

- Now the `Popover` that holds a context menu will only get set as a dialog when it isn't always on the page, it feels dishonest to have it stated as a dialog in the editor when the menu doesn't actually popup at all, it just is a menu. 
- In the `Menu` the wrapper gets the role of "menu" and its child is a "group"  (the unordered list element) and then the `li` children of the `ul` are now `role=menuitem` to show belonging to the menu instead of the list. This allows us to throw whatever we want into the list items, now that they are menuitem roles it isn't as strict of a hierarchy. 
- The component in `Menu` called `MenuItem` is a touch misleading now since the role of menuitem is given to the list item. Now the `MenuItem`  is just the children of the role=menuitem. This allows us to have buttons and anchors present in our menus dynamically, which is the major key of this whole component. 
- Added 1 aria label to the group to give context to announcements, this has a default label of `menu options`. 
- Two new checks for aXe violations - one for the quickAction menus and then while testing for regressions on the other spots the menu is used I added a second one in the dashboard myStories karma tests. Otherwise, the updates are just to the fixture and unit tests to catch up to the changes I made. 

## To-do

-N/A

## User-facing changes

- None

## Testing Instructions

- Check for regression on the quick action menu for functionality and styles, I only updated roles so I'm not anticipating any weird changes but more eyes are appreciated 
- Verify that tests are passing 
- If you want to you can also use the [aXe dev tools ](https://chrome.google.com/webstore/detail/axe-devtools-web-accessib/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US)and see that the context menu is not throwing any accessibility errors or warnings. 

### QA

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [ ] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #7751

### CODE REVIEW

**Review by samwhale — APPROVED** [review]
This is dope. Very easy to follow, and great PR description

File: assets/src/edit-story/components/canvas/karma/quickActions.karma.js [hunk 1]
```diff
@@ -43,6 +43,14 @@ describe('Quick Actions integration', () => {
     fixture.restore();
   });
 
+  describe('quick action menu should have no aXe accessibility violations', () => {
+    it('should pass accessibility tests with the default menu', async () => {
```
Comment (BrittanyIRL) [inline assets/src/edit-story/components/canvas/karma/quickActions.karma.js:47]: @swissspidy - just tagging you as reviewer so you can see how helpful what you added and suggested is! I think starting to add `toHaveNoViolations` is a really great idea. I used this in tandem with running the aXe browser tool that I've got installed on Chrome and it made for a clear update path for these menus. My team will add this to our acceptance criteria for features moving forward.

File: assets/src/design-system/components/contextMenu/menu.js [hunk 2]
```diff
@@ -167,6 +168,7 @@ const Menu = ({
   isIconMenu,
   isOpen,
   onDismiss,
+  groupLabel = __('menu options', 'web-stories'),
```
Comment (swissspidy) [inline assets/src/design-system/components/contextMenu/menu.js]:
```suggestion
  groupLabel = __('Menu options', 'web-stories'),
```

File: assets/src/edit-story/components/canvas/karma/quickActions.karma.js (hunk 1 above)
Comment (swissspidy) [reply assets/src/edit-story/components/canvas/karma/quickActions.karma.js:47]: Glad to hear that!

Are there any specific steps you followed with testing in browser + in Karma that might be worth documenting?

cc @miina

Comment (BrittanyIRL) [reply assets/src/edit-story/components/canvas/karma/quickActions.karma.js:47]: Not that I can think of at the moment, aXe is pretty comprehensive in how it describes issues and solving for them. Let's revisit after a few more instances have been added and see if there's something that would be valuable.

File: assets/src/dashboard/app/views/myStories/karma/myStories.karma.js [hunk 3]
```diff
@@ -545,14 +545,19 @@ describe('Grid view', () => {
       // now the focused item should be the first context menu item
       const [contextMenuList] =
         within(activeStoryContainer).getAllByTestId(/context-menu-list/);
+      await expectAsync(contextMenuList).toHaveNoViolations();
       // it is focused on the link within the list
       const [firstContextMenuItem, secondContextMenuItem] =
         within(contextMenuList).getAllByRole('menuitem');
-      expect(firstContextMenuItem).toEqual(document.activeElement);
+      expect(firstContextMenuItem.innerText).toEqual(
```
Comment (embarks) [inline assets/src/dashboard/app/views/myStories/karma/myStories.karma.js:552]: matching `innerText` because it wasn't matching on the `activeElement`?

**Review by embarks — APPROVED** [review]
Nice! Thanks for showing how to use the browser accessibility tools as well, it was nice running it on the dashboard and finding no accessibility issues. Let's keep it up. :)

**Pull request merged**